core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -307,12 +336,17 @@ public void onNext(ByteBuffer byteBuffer) {
             if (byteBuffer == null) {
                 throw new NullPointerException("onNext must not be called with null byteBuffer");
             }
+            if (isCancelled.get()) {
+                return;
+            }
             publisherToUpstream.send(byteBuffer).whenComplete((r, t) -> {
                 if (t != null) {
                     handleError(t);
                     return;
                 }
-                subscription.request(1);
+                if (!isCancelled.get()) {
```

Comment:
Similar question as above, is it possible this is no longer true after we check it? If so, is it an issue?

Reply:
same as above, `onStreanCalled` will only go from `false -> true` once and never back to false.

Reply:
Do we need the check here assuming subscription is cancelled properly or will be cancelled at this point? 
https://github.com/reactive-streams/reactive-streams-jvm 

> If a Subscription is cancelled its Subscriber MUST eventually stop being signaled.

Note "eventually" not "right away"

Reply:
we do need to stop right away in the case of the return future being cancelled, for example when pausing with transfer-manager

Reply:
What would happen if we don't check it here and continue to send request demand? It should be no-op right?